lib/core/proxy.c

```diff
@@ -467,15 +482,15 @@ static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errs
             if (token == H2O_TOKEN_CONTENT_LENGTH) {
                 if (req->res.content_length != SIZE_MAX ||
                     (req->res.content_length = h2o_strtosize(headers[i].value.base, headers[i].value.len)) == SIZE_MAX) {
-                    self->client = NULL;
+                    detach_client(self);
                     h2o_req_log_error(req, "lib/core/proxy.c", "%s", "invalid response from upstream (malformed content-length)");
                     h2o_send_error_502(req, "Gateway Error", "invalid response from upstream", H2O_SEND_ERROR_HTTP1_CLOSE_CONNECTION);
```

Comment:
It's become now unnecessary to close the connection, the `flag` field can be set to 0 again. This is true for the three calls to `h2o_send_error_502` in the file

Reply:
reverted by https://github.com/h2o/h2o/pull/2010/commits/e70837953b5d0f93af26499a27e135438182b61d